src/pages/core-candy-machine/guides/create-a-core-candy-machine-ui.md

```diff
@@ -0,0 +1,789 @@
+
+1. Validate the `startDate` is in the past. Note that we are not using the users devices time here but instead fetch the current internal Solana Blocktime since this is the time the Candy Machine will use for the validation on mint: 
+```ts
+import { unwrapOption } from '@metaplex-foundation/umi';
+
+let allowed = true;
+
+// fetch the current slot and read the blocktime
+const slot = await umi.rpc.getSlot();
+let solanaTime = await umi.rpc.getBlockTime(slot);
+
+// Check if a `default` startDate guard is attached
+const startDate = unwrapOption(candyGuard.guards.startDate);
+if (startDate) {
+  // validate the startTime is in the future
+  if (solanaTime < startDate) {
+        console.info(`StartDate not reached!`);
+        allowed = false;
+  }
+}
+```
+
+2. Check if the Wallet has enough SOL to pay for the mint. Note that we are not including transaction fees here and assume that the `SolBalance` was fetched as described above.
+```ts
+import { unwrapOption } from '@metaplex-foundation/umi';
+
+const solPayment = unwrapOption(candyGuard.guards.solPayment);
+if (solPayment){
+  if (solPayment.lamports.basisPoints > solBalance){
+    console.info(`Not enough SOL!`);
+    allowed = false;
+  }
+}
+```
+
+3. Make sure the `mintLimit` was not reached yet:
+```ts
+import { unwrapOption } from '@metaplex-foundation/umi';
+import { 
+  safeFetchMintCounterFromSeeds,
+} from "@metaplex-foundation/mpl-core-candy-machine";
+
+const mintLimit = unwrapOption(candyGuard.guards.mintLimit);
+if (mintLimit){
+      const mintCounter = await safeFetchMintCounterFromSeeds(umi, {
+      id: mintLimit.id,
+      user: umi.identity.publicKey,
+      candyMachine: candyMachine.publicKey,
+      candyGuard: candyMachine.mintAuthority,
+    });
+
+    // mintCounter PDA exists (not the first mint)
+    if (mintCounter && mintLimit.limit >= mintCounter.count
+    ) {
+      allowed = false;
+    }
+}
+```
+
+When the wallet is not eligible to mint it is helpful to disable the mint button and show the User the reason for it. E.g. a `Not enough SOL!` message.
+
+## Guard Routes
+Certain Guards require specific instructions to be executed before minting can occur. These instructions create accounts that store data or provide proof of a wallet's eligibility to mint. The execution frequency of these instructions varies depending on the Guard type. 
+
+{% callout type="note" title="Target Audience of this section" %}
+In case you are not using the `Allocation`, `FreezeSolPayment`, `FreezeTokenPayment` or `Allowlist` guard it is safe to skip this section.
+{% callout type="note" title="Configuration" %}
+
+Some Guards need their routes executed only once for the entire Candy Machine. For these, it's not necessary to include a function in the UI but can be run upfront once through a script:
```

Comment:
```suggestion
Some Guards need their routes executed only once for the entire Candy Machine. For these, it's not necessary to include a functional interaction (button) in the UI but can be run on page load once through a script:
```

Reply:
This is not correct from my POV. You only run it once independent of the UI.

Reply:
yep ignore me then :D